PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx

```diff
@@ -1143,6 +1143,9 @@ Double_t AliAnalysisTaskRecoilJetYield::PTD(AliEmcalJet *Jet, Int_t JetContNb){
   if(!fTruthJet) fJetInfoVar[18]=(finaljet.m());
   else fJetInfoVar[19]=(finaljet.m());
 
+
+  if(fJetDef) delete fJetDef;
+  if(recluster) delete recluster;
```

Comment:
I think both recluster and fJetDef can be constructed on the stack as the type is known at compile time and the object is not shared with other occurrences. For recluster you make a switch of the fReclusterAlgo and construct the object afterwards. Once the object is on the stack they are destructed automatically once they go out-of-scope.

Reply:
Okay it works to create on the stack for TRandom in the other case that you highlighted but for fJetDef and recluster they are passed to functions that require a type fastjet::contrib:Recluster * for example. Is it possible to do this on the stack?